src/Compiler/test/Microsoft.NET.Sdk.Razor.SourceGenerators.Tests/RazorSourceGeneratorTests.cs

```diff
@@ -2512,6 +2491,43 @@ public bool TryResolveAssemblyPaths(CompilationLibrary library, List<string>? as
             }
         }
 
+        private class RazorTest : Verify.Test
+        {
+            public RazorTest([CallerFilePath] string? testFile = null, [CallerMemberName] string? testMethod = null)
+                : base(testFile, testMethod)
+            {
+                // Don't resolve any reference assemblies from NuGet
+                ReferenceAssemblies = new ReferenceAssemblies("custom");
+
+                foreach (var defaultCompileLibrary in DependencyContext.Load(typeof(RazorSourceGeneratorTests).Assembly)!.CompileLibraries)
+                {
+                    foreach (var resolveReferencePath in defaultCompileLibrary.ResolveReferencePaths(new AppLocalResolver()))
+                    {
+                        TestState.AdditionalReferences.Add(MetadataReference.CreateFromFile(resolveReferencePath));
+                    }
+                }
+
+                // The deps file in the project is incorrect and does not contain "compile" nodes for some references.
+                // However these binaries are always present in the bin output. As a "temporary" workaround, we'll add
+                // every dll file that's present in the test's build output as a metadatareference.
+                foreach (var assembly in Directory.EnumerateFiles(AppContext.BaseDirectory, "*.dll"))
+                {
+                    if (!TestState.AdditionalReferences.Any(c => string.Equals(Path.GetFileNameWithoutExtension(c.Display), Path.GetFileNameWithoutExtension(assembly), StringComparison.OrdinalIgnoreCase)))
+                    {
+                        TestState.AdditionalReferences.Add(MetadataReference.CreateFromFile(assembly));
+                    }
+                }
```

Comment:
📝 This manually adds `MetadataReference` items for all the items that were being added to the old tests. The whole block was copied from `CreateBaseProject()`.